artemis-cli/src/main/java/org/apache/activemq/artemis/cli/commands/Upgrade.java

```diff
@@ -0,0 +1,318 @@
+    */
+   protected void checkDirectory() {
+      if (!directory.exists()) {
+         throw new RuntimeException(String.format("Could not find path '%s' to upgrade.", directory));
+      } else if (!directory.canWrite()) {
+         throw new RuntimeException(String.format("The path '%s' is not writable.", directory));
+      }
+   }
+
+
+   @Override
+   public Object execute(ActionContext context) throws Exception {
+      this.checkDirectory();
+      super.execute(context);
+
+      return run(context);
+   }
+
+
+   public Object run(ActionContext context) throws Exception {
+      File bkpFolder = findBackup();
+      File bkpBin = new File(bkpFolder, "bin");
+      File etcBkp = new File(bkpFolder, "etc");
+      bkpBin.mkdirs();
+      etcBkp.mkdirs();
+
+      File bin = new File(directory, "bin");
+      File etcFolder = new File(directory, etc);
+
+      if (!bin.exists()) {
+         throw new IOException(bin.getAbsolutePath() + " does not exist");
+      }
+
+      System.out.println("*******************************************************************************************************************************");
+      System.out.println("Upgrading broker instance " + directory + " to use artemis.home=" + getBrokerHome());
+
+      upgradeArtemis(bkpBin, bin, etcFolder);
+      upgradeArtemisCmd(bkpBin, bin, etcFolder);
+      upgradeArtemisProfile(etcBkp, etcFolder, directory);
+      upgradeBootstratp(etcBkp, etcFolder, directory);
+      upgradeLogging(etcBkp, etcFolder, directory);
+      System.out.println();
+      System.out.println("Warning: ./artemis upgrade does not bring new functionality on the script itself from newer artemis versions. Please check for any differences on artemis script in the current release!");
+      System.out.println("*******************************************************************************************************************************");
+
+      return null;
+   }
+
+   private void upgradeLogging(File bkpFolder, File etc, File instance) throws Exception {
+      File oldLogging = new File(etc, "logging.properties");
+
+      if (oldLogging.exists()) {
+         System.out.println("Moving " + oldLogging + " under " + bkpFolder);
+         File oldLoggingCopy = new File(bkpFolder, "logging.properties");
+         Files.copy(oldLogging.toPath(), oldLoggingCopy.toPath(), StandardCopyOption.REPLACE_EXISTING);
+         oldLogging.delete();
+         File newLogging = new File(etc, Create.ETC_LOG4J2_PROPERTIES);
+
+
+         if (!newLogging.exists()) {
+            System.out.println("Creating " + newLogging);
+            InputStream inputStream = getClass().getResourceAsStream("etc/" + Create.ETC_LOG4J2_PROPERTIES);
+            OutputStream outputStream = new FileOutputStream(newLogging);
+            outputStream.write(inputStream.readAllBytes());
+            inputStream.close();
+         }
+      }
+
+
+
+      File bootstrap = new File(etc, "bootstrap.xml");
```

Comment:
Why aren't you calling the function below here?

Reply:
oops... this is a copy & paste error.. I copied and pasted from bootstrap and forgot to remove it.. thanks.